asv_bench/benchmarks/benchmarks.py

```diff
@@ -422,3 +422,49 @@ def time_fillna(self, shape, limit, inplace):
             execute(self.df)
         else:
             execute(self.df.fillna(**kw))
+
+
+class BaseTimeValueCounts:
+    subset_params = {
+        "all": lambda shape: shape[1],
+        "half": lambda shape: shape[1] // 2,
+    }
+
+    def setup(self, shape, subset="all"):
+        try:
+            subset = self.subset_params.get(subset)
+        except KeyError:
+            raise KeyError(
+                f"Invalid value for 'subset={subset}'. Allowed: {list(self.subset_params.keys())}"
+            )
```

Comment:
The previous approach was more compact, why was it changed?

Reply:
it's seems more right to catch a `KeyError` while getting value from dict, instead of check that dict contain a key, and grab it after that

Reply:
Why?

Reply:
```python
assert key in dict # Checking that key in dictionary
value = dict[key]  # Getting value from a dict, which also implies
		   # checking for the existence of the key.
```
So we checked that the key in a dict for two times, why is that inconsistency? Some bad things could happen with a dict between these two lines:
- Somebody inserts some code that modifies dict
- Even if we're going to watch closely at the future reviews that these two lines always come after each other that doesn't save us from the problem that if one day this code will be used in parallel, the thread could be paused after the successful assert check and then the other thread will modify a dict, so we'll get a `KeyError` when returning to the first thread (the operation is not atomic)

This approach solves all of the problems:
```python
try:
    value = dict[key]
except KeyError:
    raise KeyError(error_explanation)
```

Reply:
ASV does not use the threading paradigm, it uses the process paradigm (and even so it does not run the calculations in parallel as it would affect the results)

Reply:
I understand that, but anyway, it's better to keep the operation atomic if it is possible (problem 1 in previous message)

Reply:
... if it is a possible and where necessary.

"Readability counts." - «The Zen of Python»